test/widgets/home_test.dart

```diff
@@ -0,0 +1,409 @@
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import 'page_checks.dart';
+import 'test_app.dart';
+
+void main () {
+  TestZulipBinding.ensureInitialized();
+
+  late PerAccountStore store;
+
+  Future<void> prepare(WidgetTester tester) async {
+    addTearDown(testBinding.reset);
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+    await store.addUsers([eg.selfUser, eg.otherUser]);
+    final stream = eg.stream();
+    await store.addStream(stream);
+    await store.addSubscription(eg.subscription(stream));
+
+    await store.handleEvent(MessageEvent(
+      id: 0, message: eg.streamMessage(stream: stream, sender: eg.otherUser)));
+
+    await tester.pumpWidget(TestZulipApp(
+      accountId: eg.selfAccount.id, child: const HomePage()));
+    await tester.pumpAndSettle();
+  }
+
+  group('bottom nav navigation', () {
+    testWidgets('preserve states when switching between views', (tester) async {
+      await prepare(tester);
+
+      check(find.byIcon(ZulipIcons.arrow_down)).findsExactly(2);
+      check(find.byIcon(ZulipIcons.arrow_right)).findsNothing();
+
+      // Collapsing the header updates inbox's internal state.
+      await tester.tap(find.byIcon(ZulipIcons.arrow_down).first);
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsExactly(2);
+
+      // Switch to channels view.
+      await tester.tap(find.byIcon(ZulipIcons.hash_italic));
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsNothing();
+
+      // The header should remain collapsed when we return to the inbox.
+      await tester.tap(find.byIcon(ZulipIcons.inbox));
+      await tester.pump();
+      check(find.byIcon(ZulipIcons.arrow_down)).findsNothing();
+      check(find.byIcon(ZulipIcons.arrow_right)).findsExactly(2);
+    });
+
+    testWidgets('update app bar title when switching between views', (tester) async {
+      await prepare(tester);
+
+      check(find.descendant(
+        of: find.byType(ZulipAppBar),
+        matching: find.text('Inbox'))).findsOne();
+
+      await tester.tap(find.byIcon(ZulipIcons.hash_italic));
+      await tester.pump();
+      check(find.descendant(
+        of: find.byType(ZulipAppBar),
+        matching: find.text('Channels'))).findsOne();
+
+      await tester.tap(find.byIcon(ZulipIcons.user));
+      await tester.pump();
+      check(find.descendant(
+        of: find.byType(ZulipAppBar),
+        matching: find.text('Direct messages'))).findsOne();
+    });
+  });
+
+  group('menu', () {
+    final designVariables = DesignVariables.light();
+
+    final inboxMenuIconFinder = find.descendant(
+      of: find.byType(SingleChildScrollView),
+      matching: find.byIcon(ZulipIcons.inbox));
```

Comment:
How about:
```suggestion
    final inboxMenuIconFinder = find.descendant(
      of: find.byType(BottomSheet),
      matching: find.byIcon(ZulipIcons.inbox));
```

That feels a bit closer to the UI as seen by the user. After all, for users not on small screens the menu doesn't even scroll.

(The point of the ancestor condition IIUC is to exclude the corresponding icon in the nav bar itself.)